rc/lsp.kak

```diff
@@ -113,40 +115,93 @@ info=$lsp_info \
     diagnostics=$lsp_diagnostics \
     code_lenses=$lsp_code_lenses \
     awk 'BEGIN {
-        info = ENVIRON["info"]
-        diagnostics = ENVIRON["diagnostics"];
-        code_lenses = ENVIRON["code_lenses"];
-        max_lines = ENVIRON["kak_opt_lsp_hover_max_lines"];
+        max_info_lines = ENVIRON["kak_opt_lsp_hover_max_lines"]
+
+        # If lsp_hover_max_info_lines is a sentanal value (e.g. -1) then it is
+        # likely the user has not set the value themselfs, use the value of the
+        # new lsp_hover_max_info_lines.
+        if (max_info_lines < 0)
+            max_info_lines = ENVIRON["kak_opt_lsp_hover_max_info_lines"]
+
+        info_lines = split(ENVIRON["info"], info_line, /\n/)
+
+        info_truncated = 0
+
+        # Will the info lines need to be truncated
+        if(is_truncated(max_info_lines, info_lines)) {
+            # Only output max_info_lines amount of info lines
+            info_lines = max_info_lines
+            info_truncated = 1
```

Comment:
> printf info_line[1]

should not use user input as format string, should be

	printf "%s" info_line[1]

I fixed it